superset/views/core.py

```diff
@@ -834,7 +834,7 @@ def explore(  # pylint: disable=too-many-locals
         }
         try:
             datasource_data = datasource.data if datasource else dummy_datasource_data
-        except (SupersetException, SQLAlchemyError):
+        except Exception:  # pylint: disable=broad-except
```

Comment:
We should try to avoid catching broad `Exception`s whenever possible. I believe there was a PR for specifying what types of exceptions are to be expected from a db engine spec under normal circumstances. Let me see if I can find it..

Reply:
I believe this is the one: https://github.com/apache/superset/pull/12869 Do you think it would be possible to use this approach here?

Reply:
Thanks! This is exactly what I'm looking for!